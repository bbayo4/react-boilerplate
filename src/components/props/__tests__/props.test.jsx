import React from 'react';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import Props from '../props';

describe('The Props Component', () => {

  const props = {
    text: 'someText',
    list: ['A', 'B', 'C'],
    func: (text) => {},
    comp: React.Component()
  };

  const wrapper = shallow(<Props {...props}/>);

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('the button calls a function', () => {
    const propsBtn = wrapper.find('button').at(0);
    propsBtn.simulate('click');
  });

});