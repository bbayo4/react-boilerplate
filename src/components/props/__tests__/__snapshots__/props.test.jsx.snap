// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Props Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Props
    func={[Function]}
    list={
      Array [
        "A",
        "B",
        "C",
      ]
    }
    text="someText"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          BASIC PROPS...
        </h1>,
        <p>
          someText
        </p>,
        <br />,
        <h3>
          A List!
        </h3>,
        <ul>
          <li>
            A
          </li>
          <li>
            B
          </li>
          <li>
            C
          </li>
        </ul>,
        <button
          onClick={[Function]}
        >
          Click me to fire off a function from a parent component
        </button>,
        <div
          className="spinner"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "BASIC PROPS...",
        },
        "ref": null,
        "rendered": "BASIC PROPS...",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "someText",
        },
        "ref": null,
        "rendered": "someText",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A List!",
        },
        "ref": null,
        "rendered": "A List!",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              A
            </li>,
            <li>
              B
            </li>,
            <li>
              C
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "A",
            },
            "ref": null,
            "rendered": "A",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "B",
            },
            "ref": null,
            "rendered": "B",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "C",
            },
            "ref": null,
            "rendered": "C",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click me to fire off a function from a parent component",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Click me to fire off a function from a parent component",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "spinner",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            BASIC PROPS...
          </h1>,
          <p>
            someText
          </p>,
          <br />,
          <h3>
            A List!
          </h3>,
          <ul>
            <li>
              A
            </li>
            <li>
              B
            </li>
            <li>
              C
            </li>
          </ul>,
          <button
            onClick={[Function]}
          >
            Click me to fire off a function from a parent component
          </button>,
          <div
            className="spinner"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BASIC PROPS...",
          },
          "ref": null,
          "rendered": "BASIC PROPS...",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "someText",
          },
          "ref": null,
          "rendered": "someText",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A List!",
          },
          "ref": null,
          "rendered": "A List!",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                A
              </li>,
              <li>
                B
              </li>,
              <li>
                C
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "A",
              },
              "ref": null,
              "rendered": "A",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "B",
              },
              "ref": null,
              "rendered": "B",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "C",
              },
              "ref": null,
              "rendered": "C",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click me to fire off a function from a parent component",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Click me to fire off a function from a parent component",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "spinner",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
